### DISCUSSION TITLE
creating two instances of teslamate for the same car_id - one for read only and the other for updates

### DISCUSSION THREAD

**Discussion by sparvata** [jheredianet/Teslamate-CustomGrafanaDashboards#56] [Q&A]
I want to create two instances of teslamate, one for viewing only (which I will populate by restoring an older database) and another instance for getting live data from my car. I was successful in creating using two instances, but seems like the view only instance cannot connect to the postgres database therefore all the dashboard queries are failing. Does anyone have any suggestions?

From the log I see the below messages. 

2023-09-24T05:47:23.807861630Z waiting for postgres at database:5432
2023-09-24T05:47:24.816184175Z waiting for postgres at database:5432
2023-09-24T05:47:25.823535559Z waiting for postgres at database:5432
2023-09-24T05:47:26.829929375Z waiting for postgres at database:5432
2023-09-24T05:47:27.837147913Z waiting for postgres at database:5432
2023-09-24T05:47:28.844702202Z waiting for postgres at database:5432
2023-09-24T05:47:29.852730782Z waiting for postgres at database:5432
2023-09-24T05:47:30.860477278Z waiting for postgres at database:5432

My docker definition for this service is like below.
  database_old:
    image: postgres:14
    restart: always
    environment:
      - POSTGRES_USER=${TM_DB_USER}
      - POSTGRES_PASSWORD=${TM_DB_PASS}
      - POSTGRES_DB=${TM_DB_NAME_OLD}
```
#    ports:
#      - "5433:5433"
```
    volumes:
      - teslamate_old-db:/var/lib/postgresql/data

**Comment by jheredianet** [comment]
Hi, I think this is the issue:
```
ports:
   - "5433:5433"
```
By default, postgres is setup to listen on port 5432, so if you wish to change the external port to avoid conflicts it should be:
```
ports:
   - "5433:5432"
```
My suggestion in any case is not to expose the database port unless you want to access outside the containers in that compose. So these lines should be commented (or deleted):
```
#ports:
#   - "5433:5432"
```

The you just need to be sure that the environment variable (.env file) should be pointing this database container:
```
TM_DB_NAME_OLD="database_old"
```

**Comment by sparvata** [comment]
thanks for your response. the ports rule was already commented in my docker file. I had two variables incorrect in my yml. The TM_DB_NAME_OLD that you pointed out was incorrect and the DB hostname was incorrect. After fixing these two I was able to connect to the psql server. Now restoring my database is causing issues with these changes which is a problem to be solved for some other day :-) Below the working version 

services:
  teslamate_old:
    image: teslamate/teslamate:1.27.3
    restart: always
    depends_on:
      - database_old
    environment:
      - ENCRYPTION_KEY=${TM_ENCRYPTION_KEY}
      - DATABASE_USER=${TM_DB_USER}
      - DATABASE_PASS=${TM_DB_PASS}
      - DATABASE_NAME=${TM_DB_NAME_OLD}
      - DATABASE_HOST=database_old
      - MQTT_HOST=mosquitto_old
      - VIRTUAL_HOST=${FQDN_TM}
      - CHECK_ORIGIN=true
      - TZ=${TM_TZ}
    ports:
      - "4001:4000"
    volumes:
      - ./import_old:/opt/app/import
    cap_drop:
      - all

  database_old:
    image: postgres:14
    restart: always
    environment:
      - POSTGRES_USER=${TM_DB_USER}
      - POSTGRES_PASSWORD=${TM_DB_PASS}
      - POSTGRES_DB=${TM_DB_NAME_OLD}
    volumes:
      - teslamate_old-db:/var/lib/postgresql/data

volumes:
  teslamate_old-db:

**Comment by jheredianet** [comment]
Great! You got it! Enjoy.
